src/crystal/scheduler.cr

```diff
@@ -123,4 +138,86 @@ class Crystal::Scheduler
     @current.resume_event.add(0.seconds)
     resume(fiber)
   end
+
+  {% if flag?(:preview_mt) %}
+    @rr_target = 0
+
+    protected def find_target_thread
+      if @@workers.empty?
+        Thread.current
+      else
+        @rr_target += 1
+        @@workers[@rr_target % @@workers.size]
+      end
+    end
+
+    def run_loop
+      loop do
+        @lock.lock
+        if runnable = @runnables.shift?
+          @runnables << Fiber.current
+          @lock.unlock
+          runnable.resume
+        else
+          @sleeping = true
+          @lock.unlock
+          oid = @worker_out.read_bytes(UInt64)
+          fiber = Pointer(Fiber).new(oid).as(Fiber)
+
+          @lock.lock
+          @sleeping = false
+          @runnables << Fiber.current
+          @lock.unlock
+          fiber.resume
+        end
+      end
+    end
+
+    def send_fiber(fiber : Fiber)
+      @lock.lock
+      if @sleeping
+        @worker_in.write_bytes(fiber.object_id)
+      else
+        @runnables << fiber
+      end
+      @lock.unlock
+    end
+
+    @@workers = [] of Thread
+
+    def self.workers
+      @@workers
+    end
+
+    def self.init_workers
+      @@workers = Array(Thread).new(worker_count) do |i|
+        if i == 0
+          worker_loop = Fiber.new(name: "Worker Loop") { Thread.current.scheduler.run_loop }
+          Thread.current.scheduler.enqueue worker_loop
+          Thread.current
+        else
+          Thread.new { Thread.current.scheduler.run_loop }
+        end
+      end
+    end
+
+    private def self.worker_count
+      env_workers = ENV["CRYSTAL_WORKERS"]?
+
+      if env_workers && !env_workers.empty?
+        workers = env_workers.to_i?
+        if !workers || workers < 1
+          LibC.dprintf 2, "FATAL: Invalid value for CRYSTAL_WORKERS: #{env_workers}\n"
```

Comment:
Why using `dprintf` instead of `STDERR.puts`, to not depend on `IO`?

Reply:
Because IO depends on Fiber, EventLoop and Scheduler, which we aren't defined. dprintf will always work, using IO will likely fail to print anything.